### ISSUE TITLE
File Compression

### ISSUE THREAD

**Issue by lukeed** [lukeed/sirv#3]
As a convenience, `sirv-cli` should compress your files on startup. This would happen only once, immediately before using `tiny-glob` to read & cache the files.

I'm 90% sure the compression would have to be file-by-file, rather than doing the entire directory at once. This is because we'd have to see if the `foobar.{hash}.gz` (or whatever) file exists first, and only produce a compressed version if it doesn't exist... otherwise `ETag` values would be wrong.

> 🤔 Maybe run `tiny-glob` _first_ after all? Then do a lookup on the returned results. Then, either rerun `tiny-glob` (since it'll still have its `CACHE` handy) or append our compressed outputs to the original set.

The `sirv` module should also take a `compress` option. It won't do any compressing (since it's the _runtime_), but it'll prefer/look to serve the compressed file over the original.

**Allowed Values:**

- [ ] `gzip` (default)
- [ ] `brotli`
- [ ] `zopfli`

**CLI Flags**

- [ ] `-c, --compress, 'Enable file compression with specified format', 'gzip'`

**Comment by cjk** [comment]
I would be interested in this feature. Is this still being worked on?
Serving compressed files would make a huge difference for me. I'm using Polka + sirv and don't want to switch to Fastify / Express just to get compressed payloads.

**Comment by GeeeCoin** [comment]
This would be great for me as well. Can't serve my gzipped js files.

**Comment by lukeed** [comment]
Available in current `@next` versions of the libs. They are toggled individually through `opts.gzip` and `opts.brotli` – or `-G/--gzip` and `-B/--brotli` for the CLI.

Decided not to include zopfli – at least at this time.

When enabled, `sirv` will try to serve a precompiled version of your file(s) to the client if they support it. So, for example, if the request for `/bundle.js` has the `Accept-Encoding: br, gzip` header, then `sirv` will look for `bundle.js.br` (if `--brotli` enabled) and then for `bundle.js.gz` (if `--gzip` enabled) before defaulting to the `bundle.js` file.

**Comment by cjk** [comment]
@lukeed This is great news - thanks for the addition!
